src/mds/inode_backtrace.cc

```diff
@@ -145,18 +145,7 @@ int inode_backtrace_t::compare(const inode_backtrace_t& other,
     if (ancestors[i].dirino != other.ancestors[i].dirino ||
         ancestors[i].dname != other.ancestors[i].dname) {
       *equivalent = false;
-      if (ancestors[i-1].version < other.ancestors[i-1].version) {
-        if (comparator > 0)
-          *divergent = true;
-        return -1;
-      } else if (ancestors[i-1].version > other.ancestors[i-1].version) {
-        if (comparator < 0)
-          *divergent = true;
-        return 1;
-      } else {
-        assert(ancestors[i-1].version == other.ancestors[i-1].version);
-        return 0;
-      }
```

Comment:
Nothing do with corruption.

The code first set 'comparator' according comparison between ancestors[i].version and other.ancestors[i].version. Then it increases variable 'i' and execute above code block. It's impossible that the '*divergent = true' line get reached.